ci-operator/config/openshift-eng/rebasebot/openshift-eng-rebasebot-main.yaml

```diff
@@ -59,6 +59,71 @@ tests:
         requests:
           cpu: 100m
           memory: 200Mi
+- as: cluster-api-provider-azure
+  cron: 0 12 * * Mon,Thu
+  steps:
+    test:
+    - as: cluster-api-provider-azure
+      commands: |
+        cat <<'EOF' > generate-manifests.sh
+        #!/bin/bash
+        set -e  # Exit immediately if a command exits with a non-zero status
+        set -o pipefail  # Return the exit status of the last command in the pipe that failed
+
+        stage_and_commit(){
+            # If commiter email and name is passed as environment variable then use it.
+            if [[ -z "$REBASEBOT_GIT_USERNAME" || -z "$REBASEBOT_GIT_EMAIL" ]]; then
+                author_flag=()
+            else
+                author_flag=(--author="$REBASEBOT_GIT_USERNAME <$REBASEBOT_GIT_EMAIL>")
+            fi
+
+            if [[ -n $(git status --porcelain) ]]; then
+                git add -A
+                git commit "${author_flag[@]}" -q -m "UPSTREAM: <drop>: Generate OpenShift manifests"
+            fi
+        }
+
+        # $REBASEBOT_SOURCE is version tag in format v0.0.0
+        if [[ -z "$REBASEBOT_SOURCE" ]]; then
+            echo "The environment variable REBASEBOT_SOURCE is not set." >&2
+            exit 1
+        fi
+
+        # Remove the leading 'v' from REBASEBOT_SOURCE
+        export PROVIDER_VERSION="${REBASEBOT_SOURCE#v}"
+
+        pushd openshift
+        make ocp-manifests
+        popd
+        stage_and_commit
+        EOF
+        chmod +x generate-manifests.sh
+
+        UPSTREAM_VERSION=$(curl --silent --header "X-GitHub-Api-Version:2022-11-28" "https://api.github.com/repos/kubernetes-sigs/cluster-api-provider-azure/releases" | grep '"tag_name":' | grep -v alpha | grep -v beta | grep -v rc | sed -E 's/.*"([^"]+)".*/\1/' | sed '/-/!{s/$/_/}' | sort -V | sed 's/_$//' | tail -n1)
```

Comment:
Also fine for now, but as I mentioned in various conversations we should have a special (optional) hook that runs before the rebase and that returns a computed source, for example:
`--source-repo=kubernetes-sigs/cluster-api-provider-azure`
`--source-revision-hook=builtin-capi-latest-stable-version.sh`, we can then retain the current `--source` behaviour to avoid breaking current use cases.

Where whatever is passed to `--source-revision-hook` must return a valid git sha/revision/tag/branch, that is then taken in by the bot and concatenated to `--source-repo` to dynamically build the source.